ethereum/datastructures/src/main/java/tech/pegasys/artemis/datastructures/networking/libp2p/rpc/BeaconBlocksByRangeRequestMessage.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.datastructures.networking.libp2p.rpc;
+
+import com.google.common.primitives.UnsignedLong;
+import java.util.List;
+import java.util.Objects;
+import org.apache.tuweni.bytes.Bytes;
+import org.apache.tuweni.bytes.Bytes32;
+import org.apache.tuweni.ssz.SSZ;
+import tech.pegasys.artemis.util.SSZTypes.SSZContainer;
+import tech.pegasys.artemis.util.sos.SimpleOffsetSerializable;
+
+public final class BeaconBlocksByRangeRequestMessage
+    implements SimpleOffsetSerializable, SSZContainer {
+
+  private final Bytes32 headBlockRoot;
+  private final UnsignedLong startSlot;
+  private final UnsignedLong count;
+  private final UnsignedLong step;
+
+  @SuppressWarnings("unused") // Required by SimpleOffsetSerializer
+  public BeaconBlocksByRangeRequestMessage() {
+    this(Bytes32.ZERO, UnsignedLong.ZERO, UnsignedLong.ZERO, UnsignedLong.ZERO);
```

Comment:
A step of zero is invalid, right?

Reply:
It is but this constructor doesn't really make any sense - and it actually turns out that because the fields are all simple types we don't actually need to have it (thought it was required to make `SimpleOffsetSerializer` happy).